command/agent/http_test.go

```diff
@@ -1006,6 +1006,12 @@ func TestHTTPServer_Limits_OK(t *testing.T) {
 			}(i)
 		}
 
+		select {
+		case err := <-errCh:
+			t.Fatalf("unexpected error from connection prior to limit: %T %v", err, err)
+		case <-time.After(500 * time.Millisecond):
+		}
```

Comment:
This block just catches test errors earlier and makes the problem clearer at the expense of 500ms per test.